backend/result/views.py

```diff
@@ -81,3 +74,24 @@ def get_result(request, question):
 
     return JsonResponse({"answer": result.given_response},
                         json_dumps_params={'indent': 4})
+
+
+def verify_session(request):
+    # Current participant
+    participant = get_participant(request)
+    # Get session for current participant
+    session_id = request.POST.get("session_id")
+
+    if not session_id:
+        return HttpResponseBadRequest("session_id not defined")
+    try:
+        session = Session.objects.get(
+            pk=session_id, participant__id=participant.id)
+    except Session.DoesNotExist:
+        return HttpResponseServerError("No session found")
+
+    # Prevent creating results when session is finished
+    if session.is_finished():
+        return HttpResponseServerError("Session has already finished")
+
+    return session
```

Comment:
Can't we make a middleware out of this and put an attribute above the `intermediate_score` method? Perhaps out of scope for now, but perhaps something worth considering.

Reply:
Perhaps a decorator will do?

Reply:
> Perhaps a decorator will do?

Yes, that's the word I was looking for!

Reply:
I decided against turning this into a decorator, as I'm not sure this would cut down the number of lines beyond a flat function call.